src/helpers/positionsOfUriInText.ts

```diff
@@ -0,0 +1,30 @@
+export interface Range {
+	start: number;
+	end: number;
+}
+
+/**
+ * Returns an array of index positions in the given string that may
+ * encapsulate URLs, or `null` if no URLs were found.
+ */
+export function positionsOfUriInText(str: string): NonEmptyArray<Range> | null {
+	const uri =
+		/https?:\/\/(www\.)?[-a-zA-Z0-9@:%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_+.~#?&//=]*)/gu;
+
+	let results: NonEmptyArray<Range> | null = null;
+	let match: RegExpExecArray | null = null;
+
+	while ((match = uri.exec(str))) {
```

Comment:
nit: extra parens? Or does this do something I'm ont aware of.

Reply:
ESLint thinks the assignment expression is accidental unless I tell it its intentional by wrapping it in parens. The goal of this line is to run the loop if a match is found. Could probs be clearer, but I didn't wanna duplicate the assignment expression lol

Thoughts on a clearer syntax?

Reply:
ah, that makes sense now